modules/oracle-xe/src/main/java/org/rnorth/testcontainers/containers/OracleContainer.java

```diff
@@ -0,0 +1,90 @@
+package org.rnorth.testcontainers.containers;
+
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerInfo;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import com.zaxxer.hikari.pool.PoolInitializationException;
+
+/**
+ * Created by gusohal on 26/04/15.
+ */
+public class OracleContainer extends AbstractContainer implements DatabaseContainer  {
+
+    private static final String IMAGE = "alexeiled/docker-oracle-xe-11g";
+    private String oraclePort;
+
+    @Override
+    protected void containerIsStarting(ContainerInfo containerInfo) {
+        oraclePort = containerInfo.networkSettings().ports().get("1521/tcp").get(0).hostPort();
+    }
+
+    @Override
+    protected String getLivenessCheckPort() {
+        return oraclePort;
+    }
+
+    @Override
+    protected ContainerConfig getContainerConfig() {
+        return ContainerConfig.builder()
+                .image(getDockerImageName())
+                .exposedPorts("1521", "22", "8080")
+                .build();
+    }
+
+    @Override
+    protected String getDockerImageName() {
+        return IMAGE + ":" + tag;
+    }
+
+    @Override
+    public String getName() {
+        return "oracle";
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "oracle.jdbc.OracleDriver";
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return "jdbc:oracle:thin:" + getUsername() + "/" + getPassword() + "@//" + dockerHostIpAddress + ":" + oraclePort + "/xe";
+    }
+
+    @Override
+    public String getUsername() {
+        return "system";
+    }
+
+    @Override
+    public String getPassword() {
+        return "oracle";
+    }
+
+    @Override
+    public void waitUntilContainerStarted() {
+        super.waitUntilContainerStarted();
+        HikariConfig hikariConfig = new HikariConfig();
```

Comment:
Ah - sorry I just noticed: please use plain JDBC for the container liveness check, not Hikari.

Much as I love Hikari and think everyone should use it for their DB connection pooling needs, I'm not quite yet ready to force it upon everybody as a compile-scoped dependency. We should be able to use regular JDBC to do the liveness check.

(NB I know the presence of the zexec library goes against the above philosophy, and I'm mulling replacing it with a JDK-standard approach!)